### PR TITLE
HDDS-5729. support setting maxSizeEnteringTarget and maxSizeLeavingSource in command line

### PR DESCRIPTION
**Pull request by JacksonYao287**
## What changes were proposed in this pull request?

add container balancer command line parameters

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5729

## How was this patch tested?

unit test

### CODE REVIEW

**Comment by JacksonYao287** [comment]
@lokeshj1703  PTAL!

**Comment by siddhantsangwan** [comment]
Note that currently ContainerBalancer doesn't support the `maxSizeLeavingSource` configuration. Only one container can move from a source datanode right now. We can add support for this config in [HDDS-5517](https://issues.apache.org/jira/browse/HDDS-5517).

**Comment by JacksonYao287** [comment]
> Note that currently ContainerBalancer doesn't support the `maxSizeLeavingSource` configuration. Only one container can move from a source datanode right now. We can add support for this config in [HDDS-5517](https://issues.apache.org/jira/browse/HDDS-5517).

ok, we can add `maxSizeLeavingSource` for now and do not set it explicitly until HDDS-5517 is merged

**Comment by siddhantsangwan** [comment]
This looks good to me!

**Review by siddhantsangwan — APPROVED** [review]

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStartSubcommand.java [hunk 1]
```diff
@@ -54,10 +54,21 @@
...
+  @Option(names = {"-e", "--maxSizeEnteringTarget"},
+      description = "the maximum size that can enter a target datanode while " +
+          "balancing in GB. This is the sum of data from multiple sources.")
+  private Optional<Long> maxSizeEnteringTarget;
+
+  @Option(names = {"-l", "--maxSizeLeavingSource"},
+      description = "maximum size that can leave a source datanode while " +
+          "balancing in GB, it is the sum of data moving to multiple targets.")
+  private Optional<Long> maxSizeLeavingSource;
+
```
Comment (lokeshj1703) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStartSubcommand.java:66]: NIT: Use InGB suffix similar to maxSizeToMovePerIterationInGB.

**Review by lokeshj1703 — APPROVED** [review]
@JacksonYao287 Thanks for working on this! The changes look good to me.

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStartSubcommand.java (hunk 1 above)
Comment (JacksonYao287) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStartSubcommand.java:66]: yea, i will do this

**Review by lokeshj1703 — APPROVED** [review]

**Comment by lokeshj1703** [comment]
@JacksonYao287 Thanks for working on this PR! @siddhantsangwan  Thanks for the review! I have committed the PR to master branch.

**Pull request merged**